src/scipp/experimental/plotting/tools.py

```diff
@@ -46,3 +37,20 @@ def fig_to_pngbytes(fig: Any):
     plt.close(fig)
     buf.seek(0)
     return buf.getvalue()
+
+
+def to_bin_edges(x, dim):
+    """
+    Convert array centers to edges
+    """
+    idim = x.dims.index(dim)
+    if x.shape[idim] < 2:
+        one = scalar(1.0, unit=x.unit)
+        return concat([x[dim, 0:1] - one, x[dim, 0:1] + one], dim)
```

Comment:
A bin width of 1 might be better? That is, change `one` to `half`?

Reply:
done